### PR TITLE
`machinelearningservice`: update sdk to 2023-04-01

### PR DESCRIPTION
**Pull request by wuxu92**
to add new resource `MachineLearningServices/registries` which need a newer api version. and update `client.Client` to use the meta client.

image

### CODE REVIEW

File: vendor/modules.txt [hunk 1]
```diff
@@ -426,9 +426,38 @@ github.com/hashicorp/go-azure-sdk/resource-manager/logic/2019-05-01/workflowtrig
...
+github.com/hashicorp/go-azure-sdk/resource-manager/machinelearningservices/2023-04-01/privateendpointconnections
+github.com/hashicorp/go-azure-sdk/resource-manager/machinelearningservices/2023-04-01/proxyoperations
+github.com/hashicorp/go-azure-sdk/resource-manager/machinelearningservices/2023-04-01/quota
+github.com/hashicorp/go-azure-sdk/resource-manager/machinelearningservices/2023-04-01/registrymanagement
+github.com/hashicorp/go-azure-sdk/resource-manager/machinelearningservices/2023-04-01/schedule
+github.com/hashicorp/go-azure-sdk/resource-manager/machinelearningservices/2023-04-01/v2workspaceconnectionresource
+github.com/hashicorp/go-azure-sdk/resource-manager/machinelearningservices/2023-04-01/virtualmachinesizes
+github.com/hashicorp/go-azure-sdk/resource-manager/machinelearningservices/2023-04-01/workspaceprivateendpointconnections
+github.com/hashicorp/go-azure-sdk/resource-manager/machinelearningservices/2023-04-01/workspaceprivatelinkresources
+github.com/hashicorp/go-azure-sdk/resource-manager/machinelearningservices/2023-04-01/workspaces
```
Comment (tombuildsstuff) [inline vendor/modules.txt:460]: we're only importing 3 of the SDK's here, are we planning on using more of these? If not, it's probably not worth using the Meta Client for this one?

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
Thanks for this PR @wuxu92 

Since we're only using 3 of the 30 SDK clients, switching to use the Meta Client and importing all 30 SDK's seems overkill - are you intending to implement additional functionality using those other SDKs anytime soon, if not we revert this to using the SDK clients directly (since it'd mean we're not importing unused dependencies).

Thanks!

File: vendor/modules.txt (hunk 1 above)
Comment (wuxu92) [reply vendor/modules.txt]: thanks for you suggestion, I'm going to use one(or two) new sdk for the new resource. I have updated this PR to use separate clients insted.

**Review by tombuildsstuff — APPROVED** [review]
LGTM - thanks for pushing that change @wuxu92

**Pull request merged**